app/assets/javascripts/admin/users/directives/resend_user_email_confirmation.js.coffee

```diff
@@ -0,0 +1,19 @@
+angular.module("admin.users").directive "resendUserEmailConfirmation", ($http) ->
+  scope:
+    email: "@resendUserEmailConfirmation"
+  link: (scope, element, attrs) ->
+    sent = false
+    text = element.text()
+    sending = " " + t "js.admin.resend_user_email_confirmation.sending"
+    done = " " + t "js.admin.resend_user_email_confirmation.done"
+    failed = " " + t "js.admin.resend_user_email_confirmation.failed"
+
+    element.bind "click", ->
+      return if sent
+      element.text(text + sending)
+      $http.post("/user/spree_user/confirmation", {spree_user: {email: scope.email}}).success (data) ->
+        sent = true
+        element.addClass "action--disabled"
+        element.text text + done
+      .error (data) ->
+        element.text text + failed
```

Comment:
Angular is good at this kind of thing! If we specify a template for the directive, we can bind variables on the isolate scope to the template, rather than using jQuery to update it manually. Something like:

```coffee
angular.module("admin.users").directive "resendUserEmailConfirmation", ($http) ->
  template: "<span>{{ resend + ' ' + status }}</span>"
  scope:
    email: "@resendUserEmailConfirmation"
  link: (scope, element, attrs) ->
    scope.resend = t "js.admin.resend_user_email_confirmation.resend"
    scope.status = ""
    sent = false
    ....
    # Then when we want to change state we can just do:
    scope.status = sending
```

and the element itself would just be:

```haml
%a{"resend-user-email-confirmation" => @user.email}
```

Reply:
Thank you Rob, I added a commit making this much easier and easier to translate as well. Great feedback.